src/meta/src/manager/catalog/fragment.rs

```diff
@@ -447,14 +448,72 @@ where
         bail!("fragment not found: {}", fragment_id)
     }
 
+    // Add the newly added Actor to the `FragmentManager`
+    pub async fn pre_apply_reschedules(
+        &self,
+        mut created_actors: HashMap<FragmentId, HashMap<ActorId, (StreamActor, ActorStatus)>>,
+    ) -> MetaResult<()> {
+        let map = &mut self.core.write().await.table_fragments;
+        for table_fragments in map.values_mut() {
+            let mut updated_actor_status = HashMap::new();
+
+            for (fragment_id, fragment) in &mut table_fragments.fragments {
+                if let Some(fragment_create_actors) = created_actors.remove(fragment_id) {
+                    for (actor_id, (actor, actor_status)) in fragment_create_actors {
+                        fragment.actors.push(actor);
+                        updated_actor_status.insert(actor_id, actor_status);
+                    }
+                }
+            }
+
+            table_fragments.actor_status.extend(updated_actor_status);
```

Comment:
We'd better not to modify `actor_status` in `pre_apply_reschedules`, because dirty status may left due to scaling failure. We should apply metadata change only after the command is executed successfully.

Reply:
I think all status here should be `Inactive`? Marking the actors inactive before running the command follows the convention of creating materialized view.

Reply:
Yes, but there's one thing that differs from creating materialized view: if the command fails, it's `TableFragments` will be removed. But in this situation, the actor status with `Inactive` will be left. Or we can add some clean logic if scale fails.

Reply:
> Or we can add some clean logic if scale fails.

Agree. We may add some logic on recovery to clean-up these invalid actors.